Zend/Zend.m4

```diff
@@ -212,6 +212,7 @@ AX_CHECK_COMPILE_FLAG([-Wlogical-op], CFLAGS="-Wlogical-op $CFLAGS", , [-Werror]
 AX_CHECK_COMPILE_FLAG([-Wformat-truncation], CFLAGS="-Wformat-truncation $CFLAGS", , [-Werror])
 AX_CHECK_COMPILE_FLAG([-Wstrict-prototypes], CFLAGS="-Wstrict-prototypes $CFLAGS", , [-Werror])
 AX_CHECK_COMPILE_FLAG([-fno-common], CFLAGS="-fno-common $CFLAGS", , [-Werror])
+AX_CHECK_COMPILE_FLAG([-pthread], CFLAGS="-pthread $CFLAGS", , [-Werror])
```

Comment:
It this change on purpose?

Reply:
yes [not much choice](https://github.com/php/php-src/runs/15258146846).

Reply:
oh now I remember why I checked for the availability of both `pthread_attr_get_np` and `pthread_attr_getstack`: The libc provides a stub for `pthread_attr_get_np`, but not for `pthread_attr_getstack`, when not linking with pthread.

There are stubs for `pthread_attr_getstackaddr` and `pthread_attr_getstacksize` though.

We should check for the availability of `pthread_attr_getstack`, or switch to `pthread_attr_getstackaddr` and `pthread_attr_getstacksize`. After that we can avoid explicitly linking to pthread in non-ZTS builds.

Reply:
Since `pthread_attr_getstackaddr` and `pthread_attr_getstacksize` are deprecated, we should go for checking the availability of `pthread_attr_getstack` and use it in `zend_call_stack_get_freebsd_pthread`.

We will compile the stub version of `zend_call_stack_get_freebsd_pthread` when not linking with pthread, which is fine.

Reply:
In fact the deprecation warning occured on Linux should I change it too there ?

Reply:
I see that you have already done it :) It's better if we can avoid depending on deprecated functions.